site/docs/guides/structured-output.mdx

```diff
@@ -0,0 +1,126 @@
+
+:::warning Deprecation Note
+The `StructuralTagsConfig` class and the `compound_grammar` field are deprecated. Use the `structural_tags_config` field on `StructuredOutputConfig` instead; it provides the same functionality and is the recommended API going forward.
+:::
+
+## Using Structured Output
+
+### Examples
+
+<LanguageTabs>
+    <TabItemPython>
+```python
+import json
+import openvino_genai as ov_genai
+
+
+pipe = ov_genai.LLMPipeline(model_path, "CPU")
+
+# Structured output configuration: simple JSON schema
+so_config = ov_genai.StructuredOutputConfig(
+    json_schema=json.dumps({
+        "type": "object",
+        "properties": {
+            "name": {"type": "string"},
+            "age": {"type": "integer"}
+        },
+        "required": ["name"]
+    }),
+)
+gen_config = ov_genai.GenerationConfig(max_new_tokens=100, structured_output_config=so_config)
+
+prompt = "Extract a person's name and age from the text and return only a JSON object.\nText: 'Alice is 29 years old.'"
+result = pipe.generate(prompt, generation_config=gen_config)
+print(json.loads(result))
+```
+    </TabItemPython>
+    <TabItemCpp>
+```cpp
+#include <openvino/genai/llm_pipeline.hpp>
+
+using namespace ov::genai;
+
+int main() {
+    std::string models_path = "/path/to/model/dir";
+    auto pipe = LLMPipeline(models_path, "CPU");
+
+    // Build a JSON schema object (represented using the SDK types) and configure structured output
+    StructuredOutputConfig so_config;
+    so_config.json_schema = R"({
+        "type": "object",
+        "properties": {
+            "name": {"type": "string"},
+            "age": {"type": "integer"}
+        },
+        "required": ["name"]
+    })";
+
+    std::string prompt = "Extract a person's name and age and return only a JSON object.\nText: 'Bob is 34 years old.'";
+
+    // Attach structured output config to a GenerationConfig and call generate
+    GenerationConfig gen_config;
+    gen_config.max_new_tokens = 100;
+    gen_config.structured_output_config = so_config;
+
+    auto result = pipe.generate(prompt, gen_config);
+
+    std::cout << "Model output: " << result.texts[0] << std::endl;
+    return 0;
+}
+```
+    </TabItemCpp>
+</LanguageTabs>
+
+### Samples and Further Reading
+
+You can find complete sample programs that demonstrate structured output generation here:
+
+- [Python sample](https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/python/text_generation/structured_output_generation.py)
+- [C++ sample](https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/cpp/text_generation/structured_output_generation.cpp)
+- [Combining multiple grammars for structured output](https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/python/text_generation/compound_grammar_generation.py)
+- [Trigger structured output during regular generation run](https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/python/text_generation/structural_tags_generation.py)
```

Comment:
There are also JS samples for structured output.
Probably we don't need to provide links to each sample here, maybe it is better to point user to samples page in docs (for C++, Python, JS), e.g. with relative link `/docs/samples/js/text_generation#7-structured-output-sample-structured_output_sample`

Reply:
Done